src/core/document.js

```diff
@@ -66,6 +66,7 @@ import { calculateMD5 } from "./calculate_md5.js";
 import { Catalog } from "./catalog.js";
 import { clearGlobalCaches } from "./cleanup_helper.js";
 import { DatasetReader } from "./dataset_reader.js";
+import { Intersector } from "./highlighted_text.js";
```

Comment:
Usually we call the file after the main content/class, so I would expect the file here to be called `intersector.js` which would also remove the ambiguity about the "highlight" term from the previous comment.